mobile-app/lib/ui/views/learn/challenge/challenge_viewmodel.dart

```diff
@@ -376,12 +378,32 @@ class ChallengeViewModel extends BaseViewModel {
     }
   }
 
+  void passChallenge(Challenge? challenge) async {
```

Comment:
What does this do?

Reply:
That function is for submitting the challenge when the user traverse to the next one. It generates the body and passes it on to the learn service where that'll then send the request to the respective endpoint.